src/libcollections/vec.rs

```diff
@@ -989,6 +989,46 @@ impl<T> Vec<T> {
             result
         }
     }
+
+    /// Splits the collection into two at the given index.
+    ///
+    /// Returns a newly allocated `Self`. `self` contains elements `[0..at]`,
+    /// and the returned `Self` contains elements `[at ..]`.
+    ///
+    /// Note that the capacity of `self` does not change.
+    ///
+    /// # Examples
+    /// ```rust
+    /// let mut vec = vec![1,2,3];
+    /// let vec2 = vec.split_off(1);
+    /// assert_eq!(vec, vec![1]);
+    /// assert_eq!(vec2, vec![2, 3]);
+    /// ```
+    #[inline]
+    #[unstable = "new API, waiting for dust to settle"]
+    pub fn split_off(&mut self, at: usize) -> Self {
+        if at >= self.len { panic!("split at larger than vec length") }
```

Comment:
Seems this would be cleaner as

assert!(at >= self.len(), "...");

The wording of the error message is also a bit awkward.